zebra-network/src/peer_set/inventory_registry.rs

```diff
@@ -283,11 +283,15 @@ impl InventoryRegistry {
 
     /// Drive periodic inventory tasks
     ///
-    /// # Details
+    /// Rotates the inventory HashMaps on every timer tick.
+    /// Drains the inv_stream channel and registers all advertised inventory.
     ///
-    /// - rotates HashMaps based on interval events
-    /// - drains the inv_stream channel and registers all advertised inventory
-    pub fn poll_inventory(&mut self, cx: &mut Context<'_>) -> Result<(), BoxError> {
+    /// Returns an error if the inventory channel is closed. Otherwise returns `Poll::Pending`, and
+    /// registers a wakeup the next time there is new inventory.
+    ///
+    /// TODO: also register a wakeup for the next timer tick.
```

Comment:
I think this is incorrect: a `Waker` gives out the ability to wake the calling task to one or more other tasks or executor functions:
> When a future is not ready yet, poll returns Poll::Pending and stores a clone of the [Waker](https://doc.rust-lang.org/std/task/struct.Waker.html) copied from the current [Context](https://doc.rust-lang.org/std/task/struct.Context.html). This [Waker](https://doc.rust-lang.org/std/task/struct.Waker.html) is then woken once the future can make progress.

https://doc.rust-lang.org/std/future/trait.Future.html#tymethod.poll

More background:
https://doc.rust-lang.org/std/task/struct.Context.html
https://doc.rust-lang.org/std/task/struct.Waker.html
https://doc.rust-lang.org/std/task/struct.Waker.html#method.wake